### PR TITLE
Killing as many 'in world' loops as feasible.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Also adds a unit test to prevent people adding more.

Not completely convinced of my own genius in regards to a number of these changes, particularly the debugging marker ones that would not reasonably be seen during general play and probably don't need to sit in memory.

### CODE REVIEW

File: code/game/objects/structures/flora.dm [hunk 1]
```diff
@@ -6,15 +6,21 @@
 	pixel_x = -16
 	layer = ABOVE_HUMAN_LAYER
 
+var/global/list/christmas_trees = list()
```
Comment (MistakeNot4892) [inline code/game/objects/structures/flora.dm]: Unsure if this is worth doing for the sake of one meme holiday proc.

File: code/modules/ZAS/Fire.dm [hunk 2]
```diff
@@ -86,6 +86,7 @@ If it gains pressure too slowly, it may leak or just rupture instead of explodin
 	zone.fire_tiles |= src
 	return 0
 
+var/global/list/hotspots = list()
```
Comment (MistakeNot4892) [inline code/modules/ZAS/Fire.dm]: This is primarily only checked during a debugging proc so unsure if it's needed.

File: code/modules/games/cards.dm [hunk 3]
```diff
@@ -17,11 +17,20 @@
 	else
 		return image(use_custom_front || deck_icon, "[card_icon]")
 
+var/global/list/card_decks = list()
```
Comment (MistakeNot4892) [inline code/modules/games/cards.dm:20]: This is another possibly unnecessary one, the only use is the missing card item.

File: code/modules/ZAS/Fire.dm (hunk 2 above)
Comment (comma) [reply code/modules/ZAS/Fire.dm]: Yeah I tihnk debugging guy can take lag, better than adding list operation to init of every fire object out there IMO

File: code/modules/games/cards.dm (hunk 3 above)
Comment (MistakeNot4892) [reply code/modules/games/cards.dm:20]: Going to keep this one, there aren't many card decks either so it's not a big imposition on memory.

**Review by comma — APPROVED** [review]

**Pull request merged**